frontend/online-shop/src/components/ProductDetail.tsx

```diff
@@ -0,0 +1,42 @@
+import { Button } from "@material-ui/core";
+import { productDetailStyle } from "../styles/ProductDetailStyle";
+
+export const ProductDetail = (props: any) => {
+  const style = productDetailStyle();
+
+  const { product } = props.location.state;
```

Comment:
As this is a component that is directly tied to a route you shouldn't use props with it. Indeed when you come to this page from the products list page you will be able to send the product and retrieve it from the props. How about when you try to access directly http://localhost:4200/products/1? Also there is a method called useParams() that can retrieve the path parameters object without using props

Reply:
You should retrieve the product id from the path and make a service call to retrieve the product with the respective id.

Reply:
I used this solution due to the fact that at the moment I did not have a service to communicate with the back end. After I finish chapter 8, the solution will look just like you said.